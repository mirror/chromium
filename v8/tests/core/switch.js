function f0() {
  switch (0) {
  }
}

print("testing f0");
f0();


function f1(x) {
  switch (x) {
    default:
      return "f1";
      break;
  }
  return "foo";
}

print("testing f1");
print(f1(0));
print(f1(1));


function f2(x) {
  var r = 0;
  switch (x) {
    case 0:
      r = "zero";
    case 1:
      r = "one";
      break;
    default:
      return "default";
  }
  return "r = " + r;
}


print("testing f2");
print(f2(0));
print(f2(1));
print(f2(2));


function f3(x, c) {
  var r = 0;
  switch (x) {
    default:
      r = "default";
      break;
    case  c:
      r = "value is c = " + c;
      break;
    case  2:
      r = "two";
      break;
    case -5:
      r = "minus 5";
      break;
    case  9:
      r = "nine";
      break;
  }
  return r;
}


print("testing f3");
print(f3( 2, 0));
print(f3(-5, 0));
print(f3( 9, 0));
print(f3( 0, 0));
print(f3( 2, 2));
print(f3( 7, 0));


function f4(x) {
  switch(x) {
    case 0:
      x++;
    default:
      x++;
    case 2:
      x++;
  }
  return x;
}

print(f4(0));
print(f4(1));
print(f4(2));

