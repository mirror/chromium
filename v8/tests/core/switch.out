testing f0
testing f1
f1
f1
testing f2
r = one
r = one
default
testing f3
two
minus 5
nine
value is c = 0
value is c = 2
default
3
3
3
